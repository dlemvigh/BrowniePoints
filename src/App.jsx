import React, { Component } from 'react';
import Person from "./Components/Person";
import './customization.scss';

const users = ["Tine", "Anders", "Daniel"]

const actions = [
  { name: "Opvask", score: 10 },
  { name: "LÃ¦s bog", score: 5},
  { name: "Skrald", score: 10 }
]

class App extends Component {
  constructor() {
    super();
    const scores = JSON.parse( localStorage.getItem("score") );
    this.state = {
      scores: scores ||   {
        Tine: 291,
        Anders: 116,
        Daniel: -19
      },
      users: {
        Tine: {
          isActive: true,
          color: "orange",
        },
        Anders: {
          color: "purple"
        },
        Daniel: {
          color: "teal"
        }
      }
    }
  }

  handleAddScore = (user, score) => {
    const newScore =  { ...this.state.scores };
    newScore[user] += score;
    this.setState({
      scores: newScore
    });
    localStorage.setItem("score", JSON.stringify(newScore));
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          { users.map(user =>
            <Person key={user} user={this.state.users[user]} name={user} score={this.state.scores[user]} actions={actions} onAddScore={this.handleAddScore} />  
          )}
        </header>
      </div>
    );
  }
}

export default App;
