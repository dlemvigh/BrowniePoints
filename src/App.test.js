import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';

it('matches snapshot', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
})

it('handle select', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.state("selected")).toEqual({});

  wrapper.instance().handleSelect("someName");
  expect(wrapper.state("selected")).toEqual({ someName: true});
})

it('add score', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.state("users").Daniel.score).toBe(0);

  const instance = wrapper.instance();
  instance.handleAddScore(10);
  expect(wrapper.state("users").Daniel.score).toBe(0);

  instance.handleSelect("Daniel");
  instance.handleAddScore(20);
  expect(wrapper.state("users").Daniel.score).toBe(20);
})