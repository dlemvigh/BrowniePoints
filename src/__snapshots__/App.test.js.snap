// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <User
      color="#F05D5E"
      key="Tine"
      name="Tine"
      onSelect={[Function]}
      score={0}
    />
    <User
      color="#1A964E"
      key="Anders"
      name="Anders"
      onSelect={[Function]}
      score={0}
    />
    <User
      color="#85C7F2"
      key="Daniel"
      name="Daniel"
      onSelect={[Function]}
      score={0}
    />
  </header>
</div>
`;
